### PR TITLE
[ISSUE #7454] Utilizing cache to avoid duplicate parsing

### LINKED ISSUE #7454: [Enhancement] Long Polling Repetitive Processing RequestHeader
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Summary

Long Polling Repetitive Processing RequestHeader

image

### Motivation

Optimize performance

### Describe the Solution You'd Like

Utilizing cache to avoid duplicate parsing

### Describe Alternatives You've Considered

No

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by guyinyou**
### Which Issue(s) This PR Fixes

Fixes #7454

### CODE REVIEW

**Comment by echooymxq** [comment]
How about add a method argument to decide cacheable？🤔️

File: remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RemotingCommand.java [hunk 1]
```diff
@@ -260,7 +261,11 @@ public void writeCustomHeader(CommandCustomHeader customHeader) {
 
     public CommandCustomHeader decodeCommandCustomHeader(
         Class<? extends CommandCustomHeader> classHeader) throws RemotingCommandException {
-        return decodeCommandCustomHeader(classHeader, true);
+        if (cachedHeader != null) {
```
Comment (joeCarf) [inline remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RemotingCommand.java:264]: 对于两次不同的入参，这种写法会出现返回结果相同的情况。符合预期吗

**Comment by guyinyou** [comment]
have changed

**Review by RongtongJin — APPROVED** [review]

**Review by joeCarf — APPROVED** [review]
LGTM

**Pull request merged**